packages/ocs/block-pool/BlockPoolDetailsPage.tsx

```diff
@@ -62,57 +59,57 @@ export const BlockPoolDetailsPage: React.FC<BlockPoolDetailsPageProps> = ({
     return (
       <Kebab
         toggleType="Dropdown"
-        launchModal={launchModal}
         extraProps={{
           resource,
           resourceModel: CephBlockPoolModel,
           namespace: CEPH_NS,
         }}
-        customKebabItems={(t) => [
+        customKebabItems={[
           {
             key: ModalKeys.EDIT_RES,
             value: t('Edit BlockPool'),
+            component: React.lazy(
+              () => import('../modals/block-pool/update-block-pool-modal')
+            ),
           },
           {
             key: ModalKeys.DELETE,
             value: t('Delete BlockPool'),
+            component: React.lazy(
+              () => import('../modals/block-pool/delete-block-pool-modal')
+            ),
           },
         ]}
       />
     );
-  }, [launchModal, resource]);
+  }, [resource, t]);
 
-  return (
-    <>
-      <Modal {...modalProps} />
-      {!loaded ? (
-        <LoadingBox />
-      ) : (
-        <DetailsPage
-          breadcrumbs={breadcrumbs}
-          actions={actions}
-          resourceModel={CephBlockPoolModel}
-          resource={resource}
-          pages={[
-            {
-              href: '',
-              name: 'Details',
-              component: BlockPoolDashboard as any,
-            },
-            {
-              href: 'yaml',
-              name: 'YAML',
-              component: YAMLEditorWrapped,
-            },
-            {
-              href: 'events',
-              name: 'Events',
-              component: EventStreamWrapped,
-            },
-          ]}
-        />
-      )}
-    </>
+  return !loaded ? (
+    <LoadingBox />
+  ) : (
```

Comment:
`DetailsPage` takes `loaded` and `loadError` props to internally display the `LoadingBox`...
```suggestion
  return 
```

Reply:
This code is from before. Let's refactor this in a different PR.

Reply:
okay...